src/Microsoft.ML.PipelineInference/AutoMlUtils.cs

```diff
@@ -15,21 +15,35 @@ namespace Microsoft.ML.Runtime.PipelineInference
 {
     public static class AutoMlUtils
     {
-        public static AutoInference.RunSummary ExtractRunSummary(IHostEnvironment env, IDataView data, string metricColumnName)
+        public static AutoInference.RunSummary ExtractRunSummary(IHostEnvironment env, IDataView result, string metricColumnName, IDataView trainResult = null)
         {
             double metricValue = 0;
+            double trainingMetricValue = -1d;
             int numRows = 0;
-            var schema = data.Schema;
+            var schema = result.Schema;
             schema.TryGetColumnIndex(metricColumnName, out var metricCol);
 
-            using (var cursor = data.GetRowCursor(col => col == metricCol))
+            using (var cursor = result.GetRowCursor(col => col == metricCol))
             {
                 var getter = cursor.GetGetter<double>(metricCol);
                 cursor.MoveNext();
                 getter(ref metricValue);
             }
 
-            return new AutoInference.RunSummary(metricValue, numRows, 0);
+            if (trainResult != null)
+            {
+                var trainSchema = trainResult.Schema;
+                trainSchema.TryGetColumnIndex(metricColumnName, out var trainingMetricCol);
```

Comment:
>trainSchema.TryGetColumnIndex(metricColumnName, out var trainingMetricCol); [](start = 16, length = 75)

I'm wondering if it's possible to tighten up this method a bit, in terms of its handling of inputs other than "happy path" inputs.

To give an example: if `metricColumnName` is not a column in `trainResult`, then `TryGetColumnIndex` will return false. However, we merely assume it succeeds, but `trainingMetricCol` will hold some undertermined value -- I guess `default(int)`. And it will happily extract that (assuming it was of type `double`).

Since these a public methods on a public class these would be `env.Check*` style checks. (If it were non-public though we'd still prefer to have at least asserts.)

Ideally we'd also make the other methods in this a bit more robust, but perhaps that's a bit beyond the scope of the PR. Might as well start here though.